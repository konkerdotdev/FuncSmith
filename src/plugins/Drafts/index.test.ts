import * as P from '@konker.dev/effect-ts-prelude';

import * as fixturesFsFm from '../../test/fixtures/fileset-frontmatter-1';
import * as unit from './index';

describe('plugins', () => {
  describe('Drafts', () => {
    it('should return the expected value', async () => {
      const pluginStack = P.pipe(P.Effect.succeed, unit.drafts());
      const actual = await P.Effect.runPromise(P.pipe([...fixturesFsFm.TEST_FILE_SET_FRONT_MATTER_1], pluginStack));
      expect(actual).toMatchSnapshot('plugin-drafts-1');
    });
  });
});
