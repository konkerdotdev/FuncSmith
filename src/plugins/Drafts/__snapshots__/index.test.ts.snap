// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins Drafts should return the expected value: plugin-drafts-1 1`] = `
[
  {
    "_id": "0000000000001111111111112222222222220001",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "link": "/a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "0000000000001111111111112222222222220021",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "p1 content",
    "date": "2024-01-01",
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": "2024-01-01",
      "draft": false,
      "layout": "foo.hbs",
      "nav": true,
      "navOrder": 1,
      "tags": [
        "tag1",
        "tag2",
      ],
      "title": "P1",
    },
    "layout": "foo.hbs",
    "link": "/posts/p1.md",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "0000000000001111111111112222222222220023",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "p3 content",
    "date": "2024-03-03",
    "fileBase": "p3",
    "fileExt": ".md",
    "fileName": "p3.md",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/posts/p3.md",
    "path": "/tmp/foo/posts/p3.md",
    "relDir": "posts",
    "relPath": "posts/p3.md",
    "title": "P3",
  },
  {
    "_id": "0000000000001111111111112222222222220024",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "p4 content",
    "date": "2024-04-04",
    "fileBase": "p4",
    "fileExt": ".md",
    "fileName": "p4.md",
    "frontMatter": {
      "date": "2024-04-04",
      "title": "P4",
    },
    "link": "/posts/p4.md",
    "path": "/tmp/foo/posts/p4.md",
    "relDir": "posts",
    "relPath": "posts/p4.md",
    "title": "P4",
  },
  {
    "_id": "0000000000001111111111112222222222220030",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "docs-index content",
    "date": "2024-11-01",
    "fileBase": "index",
    "fileExt": ".doc",
    "fileName": "index.doc",
    "frontMatter": {
      "collectionIndex": true,
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "nav": true,
      "navOrder": 2,
      "tags": [
        "tag1",
        "tag3",
      ],
      "title": "DOCS-INDEX",
    },
    "layout": "foo.hbs",
    "link": "/docs/index.doc",
    "path": "/tmp/foo/docs/index.doc",
    "relDir": "docs",
    "relPath": "docs/index.doc",
    "title": "DOCS-INDEX",
  },
  {
    "_id": "0000000000001111111111112222222222220031",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "d1 content",
    "date": "2024-11-01",
    "fileBase": "d1",
    "fileExt": ".doc",
    "fileName": "d1.doc",
    "frontMatter": {
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "title": "D1",
    },
    "layout": "foo.hbs",
    "link": "/docs/d1.doc",
    "path": "/tmp/foo/docs/d1.doc",
    "relDir": "docs",
    "relPath": "docs/d1.doc",
    "title": "D1",
  },
  {
    "_id": "0000000000001111111111112222222222220032",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "d2 content",
    "date": "2024-11-02",
    "fileBase": "d2",
    "fileExt": ".doc",
    "fileName": "d2.doc",
    "frontMatter": {
      "date": "2024-11-02",
      "layout": "bar.hbs",
      "title": "D2",
    },
    "layout": "bar.hbs",
    "link": "/docs/d2.doc",
    "path": "/tmp/foo/docs/d2.doc",
    "relDir": "docs",
    "relPath": "docs/d2.doc",
    "title": "D2",
  },
  {
    "_id": "0000000000001111111111112222222222220033",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "d3 content",
    "date": "2024-11-03",
    "fileBase": "d3",
    "fileExt": ".doc",
    "fileName": "d3.doc",
    "frontMatter": {
      "date": "2024-11-03",
      "title": "D3",
    },
    "link": "/docs/d3.doc",
    "path": "/tmp/foo/docs/d3.doc",
    "relDir": "docs",
    "relPath": "docs/d3.doc",
    "title": "D3",
  },
  {
    "_id": "0000000000001111111111112222222222220034",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "d4 content",
    "date": "2024-11-04",
    "fileBase": "d4",
    "fileExt": ".doc",
    "fileName": "d4.doc",
    "frontMatter": {
      "collectionExclude": true,
      "date": "2024-11-04",
      "title": "D4",
    },
    "link": "/docs/d4.doc",
    "path": "/tmp/foo/docs/d4.doc",
    "relDir": "docs",
    "relPath": "docs/d4.doc",
    "title": "D4",
  },
]
`;
